api/v1alpha1/dataimportjob_types.go

```diff
@@ -0,0 +1,141 @@
+// everest-operator
+// Copyright (C) 2022 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:shortName=dij
+//+kubebuilder:printcolumn:name="TargetCluster",type="string",JSONPath=".spec.targetClusterName"
+//+kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase"
+
+// DataImportJob is the schema for the dataimportjobs API.
+type DataImportJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataImportJobSpec   `json:"spec,omitempty"`
+	Status DataImportJobStatus `json:"status,omitempty"`
+}
+
+type DataImportJobSpec struct {
+	// TargetClusterName is the reference to the target cluster.
+	// +kubebuilder:validation:Required
+	TargetClusterName      string `json:"targetClusterName,omitempty"`
+	*DataImportJobTemplate `json:",inline"`
+}
+
+// DataImportJobTemplate defines a shared template for the data import job.
+type DataImportJobTemplate struct {
+	// DataImporterName is the data importer to use for the import.
+	// +kubebuilder:validation:Required
+	DataImporterName string `json:"dataImporterName,omitempty"`
+	// Source is the source of the data to import.
+	// +kubebuilder:validation:Required
+	Source *DataImportJobSource `json:"source,omitempty"`
+	// Config defines the configuration for the data import job.
+	// These options are specific to the DataImporter being used and must conform to
+	// the schema defined in the DataImporter's .spec.config.openAPIV3Schema.
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	Config *runtime.RawExtension `json:"config,omitempty"`
+}
+
+type DataImportJobSource struct {
+	// S3 contains the S3 information for the data import.
+	// +optional
+	S3 *DataImportJobS3Source `json:"s3,omitempty"`
+	// Path is the path to the directory to import the data from.
+	// This may be a path to a file or a directory, depending on the data importer.
+	// +optional
+	Path string `json:"path,omitempty"`
```

Comment:
Does it make sense to include this already if we are not planning to create an importer that consumes this? Shouldn't we add this when we actually need it? Respecting YAGNI.

Reply:
Why won't we need it even now? 🤔 

Even with the operator based importers, the user needs to provide a path to where the backup directory is stored in a bucket. The data importer has no way of knowing or guessing this. Am I missing something?